skimage/util/process.py

```diff
@@ -0,0 +1,68 @@
+from multiprocessing import cpu_count
+
+import dask.array as da
+
+__all__ = ['process_chunks']
+
+
+def _get_chunks(shape, ncpu):
```

Comment:
@jni the docs @mrocklin linked to explain it better than I can.

Currently `_get_chunks` returns a sequence of chunksizes (chunk and block are synonymous) along each dimension. This way we can specify what happens at the boundaries to make sure they are not smaller than the ghosting depth.

If `_get_chunks` just returned one chunk's dimensions so that this chunk was tiled across the array (this is what you are suggesting I think), the chunks on the boundaries could be truncated so that the ghosting depth might end up larger than a chunk. Which causes issues.